### PR TITLE
create file interface

### PR DESCRIPTION
**Pull request by Grace-xh**

### CODE REVIEW

File: service/file_controller_create_volume.go [hunk 1]
```diff
@@ -186,3 +206,68 @@ func queryShareData(conn RestEndpoint, fsName string) (map[string]string, error)
...
+		jsonParsed, _ := gabs.ParseJSON([]byte(resp))
+		content := jsonParsed.Path("content").Data().(float64)
+
+		if int(content) != 1 {
+			logrus.Error("Error: failed to create file interface for nas server %s", fileIPInput.nasServer)
+			err = errors.New(fmt.Sprintf("Error: failed to create file interface for nas server:%s", fileIPInput.nasServer))
+		}else{
+			children, _ :=jsonParsed.S("content").Children()
+			for _, child := range children{
+				fileInterfaceData["id"] =  child.Path("id").Data().(string)
```
Comment (jicahoo) [inline service/file_controller_create_volume.go]: Forgot to return fileInterfaceData

**Review by jicahoo — APPROVED** [review]
Looks great

File: service/file_controller_create_volume.go [hunk 2]
```diff
@@ -186,3 +206,68 @@ func queryShareData(conn RestEndpoint, fsName string) (map[string]string, error)
...
+
+func createFileInterface(conn RestEndpoint, fileIPInput fileInterfaceInput)(map[string]string, error) {
+	var err error = nil
+	fileInterfaceData := make(map[string]string)
+
+	fileInterfaces, _ := queryFileInterface(conn, fileIPInput.nasServer)
+	if fileInterfaces.Len() == 0 {
+		logrus.Info("Info: file interface is not created for nas server %s", fileIPInput.nasServer)
+		// start to create interface
+		var createFileInterfaceUrl string = "/api/types/fileInterface/instances"
```
Comment (qingfeng0820) [inline service/file_controller_create_volume.go]: I suggested to use async mode for file interface creation by adding timeout=0.
In sync mode, If the create request last a long time waiting and there is a network issue happened, you may lose the created file interface id.
If you use async mode, you can get a job id quickly. Then you can use this job id to get the creation result any time.

**Review by jicahoo — APPROVED** [review]

File: service/file_controller_create_volume.go (hunk 2 above)
Comment (Grace-xh) [reply service/file_controller_create_volume.go]: Thanks for comments, updated with async call in new commit.

File: service/file_controller_create_volume.go (hunk 1 above)
Comment (Grace-xh) [reply service/file_controller_create_volume.go]: intended to forget, haha. add it in latest commit for future use. Thanks for review and comments.

**Pull request merged**